lib/polling-consumer.js

```diff
@@ -110,9 +111,10 @@ async function pollForRecords(instance) {
       return;
     }
 
-    if (seqNumToCheckpoint) {
-      await setCheckpoint(seqNumToCheckpoint);
+    if (seqNumToCheckpoint && timestamp) {
```

Comment:
we shouldnt check for a timestamp here, i.e. even if there is no timestamp we still want to checkpoint. not saving a checkpoint because there is no timestamp could cause issues for existing users of this module.